Config validation

ftassi opened the pull request:
This adds some validation to the Config object inner data. Have a look at `\Idephix\ConfigTest` in particular for the messages returned to the user.

This combined to a more complete and expressive `idxrc.php` (with all the defaults commented) should be enough to ease ideaphix setup

micheleorselli commented:
Some files are checked in the bin directory, like 'box', 'phpunit', 'php-cs-fixer'. Was it intended? Afaik they should be excluded since they are managed by composer

ftassi commented:
See 102ae98 I removed some deps from composer and switched to phar to avoid some dependencies issues. The other solution would have been to define the exact version in the composer.json file but since they are all dev deps I preferred the phar

micheleorselli commented on docs/Cookbook/Deploy/idxrc.php:
```diff
@@ -19,9 +24,9 @@
-        \Idephix\Config::SSHCLIENT => new \Idephix\SSH\SshClient(new \Idephix\SSH\CLISshProxy()),
-        \Idephix\Config::EXTENSIONS => array(
+        'envs' => $environments,
+        'ssh_client' => new \Idephix\SSH\SshClient(new \Idephix\SSH\CLISshProxy()),
```
is `CLISshProxy` an optional dependency? Asking since in the doc seems it's not needed

ftassi replied:
Yes, it is optional indeed: SshClient will create a CliSshProxy instance in null is passed

micheleorselli replied:
Maybe we can remove it for clarity
